test/tstKokkosToolsAnnotations.cpp

```diff
@@ -43,15 +39,14 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(bvh_bvh_allocations_prefixed, DeviceType,
   Kokkos::Tools::Experimental::set_allocate_data_callback(
       [](Kokkos::Profiling::SpaceHandle /*handle*/, char const *label,
          void const * /*ptr*/, uint64_t /*size*/) {
-        BOOST_TEST_MESSAGE(label);
-        BOOST_TEST(
-            (isPrefixedWith(label, "ArborX::BVH::") || // data member
-             isPrefixedWith(label, "ArborX::BVH::BVH::") ||
-             isPrefixedWith(label, "ArborX::Sorting::") ||
-             isPrefixedWith(label, "Kokkos::SortImpl::BinSortFunctor::") ||
-             isPrefixedWith(label,
-                            "Kokkos::Serial::") || // unsure what's going on
-             isPrefixedWith(label, "Testing::")));
+        std::regex re("^(Testing::"
+                      "|ArborX::BVH::"
+                      "|ArborX::Sorting::"
+                      "|Kokkos::SortImpl::BinSortFunctor::"
+                      "|Kokkos::Serial::" // unsure what's going on
+                      ").*");
+        BOOST_TEST(std::regex_match(label, re),
+                   "\"" << label << "\" matches the regular expression");
```

Comment:
Adding this, I see something like
```
/tmp/ArborX/test/tstKokkosToolsAnnotations.cpp:48: error: in "KokkosToolsAnnotations/bvh_bvh_allocations_prefixed<Kokkos__Device<Kokkos__Threads_ Kokkos__HostSpace>>": "ArborX::BVH::internal_and_leaf_nodes" matches the regular expression
```
for failing tests which I would rather have as an error message:
```
Users/6da/Software/ArborX/test/tstKokkosToolsAnnotations.cpp:48: error: in "KokkosToolsAnnotations/bvh_bvh_allocations_prefixed<Kokkos__Device<Kokkos__Threads_ Kokkos__HostSpace>>": "ArborX::BVH::internal_and_leaf_nodes" doesn't match the regular expression
```
```suggestion
                   "\"" << label << "\" doesn't match the regular expression");
```
(given that `error` is the default log level).

Reply:
Done